src/components/matrixSize.jsx

```diff
@@ -17,10 +17,33 @@ export default class MatrixSize extends React.Component {
       this.setState({ numCols: event.target.value });
   }
   handleClick(){
-  firebase.database().ref('grids/' + this.state.gridId + '/numRows' ).set(parseInt(this.state.numRows));
-  firebase.database().ref('grids/' + this.state.gridId + '/numCols' ).set(parseInt(this.state.numCols));
+  firebase.database().ref('grids/' + this.state.gridId).once('value', gridSnapshot => {
+      var oldNumRows = gridSnapshot.child('numRows').val();
+      var oldNumCols = gridSnapshot.child('numCols').val();
 
-   this.props.updateGrid(this.state.gridId);
+      if(oldNumRows !== null && oldNumCols !== null && (oldNumRows > this.state.numRows || oldNumCols > this.state.numCols))
+      {
+          for(var rowToDelete = this.state.numRows; rowToDelete < oldNumRows; rowToDelete++)
```

Comment:
i am deleting all the rows between the new number they specify and the old number